[BE 리뷰요청] 숙소예약 프로젝트 1주차 리뷰 요청합니다.

sigridjineth opened the pull request:
### 리뷰 요청사항
* 이번 주는 전반적으로 프로젝트 킥오프 하는 단계라서 많이 작업이 되지 않았습니다.
* 이번에는 3주라서 매주 진행사항을 리뷰 요청드리고자 합니다.
* 현재까지 구현한 기능은 슬라이드 2에서 요구하는 전체 정보 반환하는 API입니다.
* ERD, Schema 등에 대한 리뷰도 요청드립니다.
* 이번에는 DDD를 의식하면서 JPA를 사용했습니다. 코드를 보시고 DDD/JPA 학습에 필요한 피드백을 기다리겠습니다!

### ERD
![image](https://user-images.githubusercontent.com/41055141/82641481-5b80cc00-9c47-11ea-806c-1e0444a8694f.png)

### 질문 사항
* DDD에서는 스키마를 먼저 짜지 않고, 클래스 다이어그램부터 그린다는데 어떻게 해야할까요?
* DDD를 의식하면서 패키지 구조를 만들었는데, 현재 짜 놓은 구조가 어떠한가요?
* RoomType에서 메타 데이터화 한 코드를 활용하여 Room 엔티티에서 가져올 때 ```@Enumerated(EnumType.Ordinal)```을 이용했는데 Ordinal을 어쩔 수 없이 사용한 부분이 있는데요, 더 나은 방법이 있을까요?

wooody92 committed:
클래스 별 디렉토리 생성

sigridjineth committed:
chore: project initalization

wooody92 committed:
[#24] feat : 기본 Entity 생성

- 테이블에 맞는 Entity 생성했습니다.

wooody92 committed:
[#24] feat : 기본 Repository 생성

- 테이블에 맞는 Repository 생성했습니다.

wooody92 committed:
[#24] feat : MySQL 데이터 출력

- MySQL 연동하여 저장되어있는 데이터 출력했습니다.

wooody92 committed:
[#24] feat : Host Entity에 superhost 컬럼 추가

- HostRepository.java : superhost 컬럼 추가했습니다. default 값 설정 추가해야 합니다.

wooody92 committed:
[#24] test : Host save하는 테스트 코드 작성

- RepositoryTest.java : Host 객체를 정상적으로 저장하는지 확인하는 테스트 코드 작성했습니다.

wooody92 committed:
[#24] feat : RoomResponseDto 생성

- RoomResponseDto.java : 메인화면의 숙소정보를 나타내는 dto 생성했습니다.

sigridjineth committed:
test: add RoomRepository and HostRepository find method to RepositoryTest

sigridjineth committed:
[#24] feat: mapping Host and Room entity with one-to-many relationship

- in order for avoiding both-side references error, add @JsonIgnore annotation to Host's room.

sigridjineth committed:
[#24] feat: mapping Room and Image entity with many-to-one relationship

sigridjineth committed:
[#26] refactor: applying DDD-principle oriented package structure

sigridjineth committed:
[#24] test: attest room to have multiple images

sigridjineth committed:
[#24] feat: mapping Room-Image to have @OneToMany relationship

- by using @ToString(exclude = "images"), avoiding stackoverflow error

sigridjineth committed:
[#24] fix: excluding rooms field in Host when toString-ing

sigridjineth committed:
[#24] feat: RoomService to return all rooms as RoomResponseDto

sigridjineth committed:
[#24] test: RepositoryTest to attest findAll() method working well

sigridjineth committed:
[#24] feat: add enum RoomType for Apartment, House and Office

sigridjineth committed:
[#24] feat: Room to have EnumType field

- TODO: pondering about refactoring not to use EnumType.ORDINAL

sigridjineth committed:
[#24] feat: RoomResponseDto to return RoomType and Image

sigridjineth committed:
[#24] chore: TestController to return RoomResponseDtos from roomService

sigridjineth committed:
[#24] chore: fix location

sigridjineth committed:
[#24] chore: change location

sigridjineth committed:
Merge pull request #31 from codesquad-member-2020/BE/feat/entity

sigridjineth committed:
[#24] feat: RoomResponseDto to have superhost field

sigridjineth committed:
[#24] chore: mock data as csv

sigridjineth committed:
Merge pull request #32 from codesquad-member-2020/BE/feat/entity

sigridjineth committed:
[#19] chore: add dependency of flyway

sigridjineth committed:
[#24] chore: properties to adopt flyway configuration

sigridjineth committed:
[#24] chore: csv files into sql

wheejuni commented on BE/csv/mock_host.csv:
```diff
@@ -0,0 +1,20 @@
+1,JK,jk@gmail.com,0
```
이메일은 그냥 가공의 이메일인 것이죠? 혹시나 해서...
GitHub에 푸시할 파일을 작성할 땐 늘 조심하시고요.

sigridjineth replied:
설마 저런 이메일을 쓰시겠어요 그냥 닉네임에 넣었습니다

wheejuni commented on BE/csv/mock_room.csv:
```diff
@@ -0,0 +1,50 @@
+27,롯데캐슬골드,0,1470000,8,8,3,3,4.78,300,강남구,127.100592,37.5144115,서울특별시 송파구 신천동 7-18,20
+28,시그니엘,0,2000000,5,4,3,3,4.89,200,강남구,127.104765,37.513579,서울특별시 송파구 잠실6동 올림픽로 300,19
+29,백현동 판교푸르지오그랑블,0,1500000,6,3,8,8,3.78,200,영등포구,126.910458,37.5198512,서울특별시 영등포구 영등포동2가 213,18
+30,정자동 분당파크뷰,0,1450000,5,5,7,7,4.89,100,영등포구,126.900423,37.4872268,서울특별시 영등포구 대림동 1095-35,2
```
분당구 금곡동은 왜 없죠?

sigridjineth replied:
오 내집 마련을 저기에?🤭

wooody92 replied:
ㅋㅋㅋㅋㅋㅋㅋㅋ

wheejuni commented on BE/src/main/java/dev/codesquad/airbnb02/Airbnb02Application.java:
```diff
@@ -0,0 +1,15 @@
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+
+@SpringBootApplication
+@EntityScan
```
`@EntityScan` 애노테이션 사용해본 것은 좋은데요,
이 애노테이션의 역할 범위와 기능을 고려했을 때, 이 클래스에 선언되는 게 적절할지는 한번만 더 고민해주세요.

wheejuni commented on BE/src/main/java/dev/codesquad/airbnb02/application/controller/TestController.java:
```diff
@@ -0,0 +1,30 @@
+@RestController
+public class TestController {
+
+    private final HostRepository hostRepository;
```
`Repository` 를 View layer에서 직접 주입받아도 괜찮은 걸까요?
아직 service 클래스를 구성할만큼의 복잡도가 생기지 않았나보네요.

sigridjineth replied:
당연히 Service 객체에서 주입 받아야 하겠으나 현재는 테스트로 일단 레포지토리 주입을 해둔 상태입니다.

wheejuni replied:
역시 제가 괜한 걱정을....

wheejuni commented on BE/src/main/java/dev/codesquad/airbnb02/application/controller/TestController.java:
```diff
@@ -0,0 +1,30 @@
+    private RoomRepository roomRepository;
+
+    @Autowired
+    private RoomService roomService;
```
`RoomRepository` 와 `RoomService` 를 동시에 한 클래스에서 주입받는 것은 기능상의 중복입니다. 
`RoomService` 에 위임해야 할 기능을 컨트롤러에서 수행하고 있지는 않은지 다시 한 번 고민해주세요.

wheejuni commented on BE/src/main/java/dev/codesquad/airbnb02/application/dto/ReviewDto.java:
```diff
@@ -0,0 +1,22 @@
+        this.count = count;
+    }
+
+    public static ReviewDto create(Double rating, Integer count) {
```
`create()` 보다 더 나은 메소드명 고민해봅니다.
보통 DTO에는 `of` 나 `with` 같은 suffix를 즐겨 사용합니다. 
`ofRatingAndCount()` 정도로...

sigridjineth replied:
이펙티브 자바에서 봤을 때 create라는 예시를 봤어서 습관적으로 create을 사용했는데요. 말씀해주신 것 들어보니 ```ofRatingAndCount()```처럼 더 명시적으로 주는 것도 좋겠네요.

wheejuni commented on BE/src/main/java/dev/codesquad/airbnb02/application/dto/RoomResponseDto.java:
```diff
@@ -0,0 +1,48 @@
+  private ReviewDto review;
+  private boolean superhost;
+
+  public RoomResponseDto() {
+  }
```
**한 줄로**

wheejuni commented on BE/src/main/java/dev/codesquad/airbnb02/application/dto/RoomResponseDto.java:
```diff
@@ -0,0 +1,48 @@
+    this.superhost = superhost;
+  }
+
+  public static RoomResponseDto create(Room room) {
+    return new RoomResponseDto(room.getId(), room.getName(), room.getType().getType(), room.getLocation(),
+        room.getImages().stream().map(Image::getImageUrl).collect(Collectors.toList()), room.getPrice(), false,
+        ReviewDto.create(room.getReviewRating(), room.getReviewCount()), room.getHost().isSuperhost());
+  }
```
워후 엄청나네요 이럴땐 `@Builder` 사용하시고, 스태틱 팩토리 메소드 안에서도 builder를 사용하는 것이 낫습니다.

sigridjineth replied:
빌더 패턴이 아무튼 디자인 패턴이라 최대한 사용을 지양하려고 했는데 이제는 써도 되려나요🤔

wheejuni replied:
네 이제는 마음껏 사용하셔요. 
사실 다른분들은 진작부터 사용하고 계신.... ㅠㅠ

wheejuni commented on BE/src/main/java/dev/codesquad/airbnb02/domain/host/data/HostRepository.java:
```diff
@@ -0,0 +1,8 @@
+import dev.codesquad.airbnb02.domain.host.entity.Host;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface HostRepository extends JpaRepository<Host,Long> {
```
콤마 뒤에 한 칸 띄웁니다. 
```suggestion
public interface HostRepository extends JpaRepository<Host, Long> {

wheejuni commented on BE/src/main/java/dev/codesquad/airbnb02/domain/host/entity/Host.java:
```diff
@@ -0,0 +1,51 @@
+@Entity
+@Getter
+@Setter
+@ToString(exclude = "rooms")
```
롬복 매니아시네요. `exclude` 를 알다니

sigridjineth replied:
이게 JPA에서 Host - Room 사이에 양방향 순환참조 문제가 나타나길래 임의로 제거해준 것입니다. 그런데 이렇게 순환참조를 빼주는 것이 임시방편 같다는 생각이 계속 들어요.

wheejuni replied:
이런 순환참조 문제때문에 DTO 클래스를 굳이 작성해서 view layer - domain layer간 상호 종속성을 끊으려는 이유도 있어요. 그치만 말씀처럼 `ToString` 이 마지막 복병인데요.
지금처럼 JPA 양방향 연관관계 매핑을 할 때는 피할 수 없는 문제라 이렇게 인위적으로 순환참조가 발생하는 메소드나 필드를 빼주는 게 유일한 해결책이에요. 임시방편 아니라서 걱정 안하셔도 됩니다. 
그러나 본질적으로는... 양방향 연관관계 매핑이 필요한지는 늘 고민하는 것이 좋습니다. 사실 필요하지 않은 경우도 많거든요.

sigridjineth replied:
어디선가 양방향 관계를 쓰지 마라는 이야기를 많이 들은 것 같은데요. 예를 들어서 Room이 있고 Room의 Location 테이블이 있으면 1:1이겠죠. 아마 서로의 id primary key로 공유될 것 같습니다. 그런데 테이블을 정규화할 것이면 이렇게 양방향으로 관계를 맺는 것이 자연스럽지 않나요? 다른 이유 OR 대안이 있을까요?

sigridjineth replied:
그리고 ```host-room```은 entity 간의 관계인데, view-domain layer 간의 DTO 클래스를 사용하여 해결하려고 한다는 것이 어떤 말씀인지 잘 이해가 되지 않습니다 ㅇㅁㅇ

wheejuni replied:
항상 자연스럽지는 않죠. `Location` 만으로 `Room` 을 특정지어야 할 이슈가 있을까요? 
정규화했다고 해서 JPA상의 양방향 연관관계 매핑이 꼭 수립돼야 할 필요가 있는 건 아니잖아요. `Location` 클래스에 `Room` 필드가 꼭 있어야 하는 걸까요?

wheejuni replied:
DTO를 통해 필요한 정보만 고립시키면 entity class에서 굳이 순환참조가 발생할만한 메소드를 작성할 필요가 없어지니까요.

sigridjineth replied:
@wheejuni 납득이 되네요. Location은 Room의 정보를 꼭 알고 있을 이유는 없겠네요. 양방향 연관관계를 지양해야 하는 이유가 있을까요? 물론, 연관관계 자체는 적게 맺는 것이 좋지 않을까 하는 개인적인 생각은 있습니다. 데이터 validity가 엄청 중요한 것이 아닌 이상은요. (그래서 N:N은 최대한 안 쓰려고 합니다.)

sigridjineth replied:
@wheejuni 아, 그렇다면 상호 종속적인 필드의 경우에는 DTO를 따로 Mapper에서 받아 생성한 후 이를 entity에 주입하는 방식이 될까요? 현재로서는 생각나는 경우가 이것이네요.

wooody92 replied:
@wheejuni 
양방향 연관관계에서 순환참조 문제때문에 위 방식처럼 `@ToString(exclude = "rooms")` , `JsonIgnore` 처리해줬습니다. 여기서 1:N 연관관계의 주인이 아닌 부분(외래키를 갖고있지 않는, `mappedBy`를 사용하는 객체) 쪽의 필드값을 Ignore 처리 해줬는데, 연관관계의 주인이 아닌 쪽에 Ignore 처리 해주는 것이 맞는 방식일까요? 어떻게 생각하시는지 궁금합니다.

wheejuni commented on BE/src/main/java/dev/codesquad/airbnb02/domain/room/entity/RoomType.java:
```diff
@@ -0,0 +1,23 @@
+package dev.codesquad.airbnb02.domain.room.entity;
+
+public enum RoomType {
+  APARTMENT(0, "아파트"),
+  HOUSE(1, "단독주택"),
+  OFFICE(2, "사무실");
```
이렇게 예쁘게 enum을 잘 설계했는데 굳이 `EnumType.ORDINAL` 로 매핑할 이유는 없었을 것 같은데요,
가급적이면 `EnumType.STRING` 으로 해 주세요. 혹시 다른 이유가 있나요?

sigridjineth replied:
오디널을 쓰지 말아야 하는 이유가 순서가 바뀔 수도 있으니 그런 것일까요?
제가 작업했을 때는 DB 상에서 0, 1, 2로 들어가 있어서 String이 아니기에 EnumType.STRING을 활용할 수 없더군요. DB에 저장된 코드(0, 1, 2)를 활용하여 이넘 타입 코드와 비교하여, 타입을 가져오도록 작업하고 싶은데 아직 숙달되지 않아 임시로 오디널로 작업해 둔 상태입니다. 좋은 방법이 있을까요?

wheejuni replied:
지금 schema 상에서 보면 `type INT` 로 선언되어 있는데 이 부분이 `type VARCHAR(32)` 정도면 `EnumType.STRING` 사용할 수 있으리라 보여져요. 

말씀하신 구조로 가려면 `AttributeConverter` 라는 것을 사용해서 별도의 구현이 필요한 부분이 있겠네요. 그런데 추천하고 싶진 않아요. DBMS 상에서도 명시적으로 타입을 볼 수 있으면 좋지 않을까요.

sigridjineth replied:
아 그렇군요. ```0, 1, 2```를 ```VARCHAR(32)```로 넣으면 ```"0", "1", "2"``` 로 인식하나 보군요!

wheejuni replied:
아뇨;; 물론 뭐 물리적으로 그렇긴 하겠지만;;
제 말씀은 `VARCHAR(32)` 로 바꾸고 `EnumType.STRING` 을 사용하시란 말씀이었어요.

sigridjineth replied:
사실 이게 메타데이터화와 연결되는 고민거리가 됩니다. 지난번에 Dan이랑 프로젝트를 하면서 메타 데이터화(?)가 익숙해졌는데요.
굳이 ```0, 1, 2```로 구분할 필요가 있느냐 - 아파트면 아파트라고 ```VARCHAR(32)```로 넣으면 되지 - 라는 주장이 있는 한편, 메타 데이터를 주장하는 측에서는 아파트의 이름이 변경될 수도 있지 않느냐 - 라는 의견이 있는 것 같습니다. 생각해보면 이번에는 굳이 메타 데이터화 하지 않고 그냥 이름으로 넣어도 괜찮을 것 같네요. 이유는 엥간하면 불변하니까요.

sigridjineth replied:
슬랙 메시지가 날라가는 것이 우려되므로, 위와 관련된 브라이언님의 메시지를 기록해두겠습니다.

```
enum의 이름이 비즈니스 결정에 따라 바뀌어야 하는 설계 자체에 대한 취약성을 갖고 있다.
그런 이름이라면 하위 필드로서 문자열로 들고 있는 것이 좋겠다.
만약 아파트가 아파트먼트로 바뀌어야 한다면 그 이넘은 이름 자체가 아파트이면 안되는 거였단 것이다. HOUSE_TYPE_A 정도로 더 추상화했어야 한다.
이 정도의 추상화 수준을 결정하는 것은 개발 당시에 충분히 산정할 수 있는 이슈라고 보여진다.
```

wheejuni commented on BE/src/main/java/dev/codesquad/airbnb02/domain/room/entity/Image.java:
```diff
@@ -0,0 +1,36 @@
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private Long id;
+
+  @NotNull
```
`@NotNull` 애노테이션 잘못 쓰신 것 같은데요.
임포트하실 때 항상 패키지 확인하세요.

wheejuni commented on BE/src/main/java/dev/codesquad/airbnb02/domain/room/entity/Room.java:
```diff
@@ -0,0 +1,83 @@
+  private String name;
+
+  @NotNull
+  @Enumerated(EnumType.ORDINAL)
```
`RoomType` 에서 말씀드린 바와 같이, 굳이 `EnumType.ORDINAL` 사용할 이유 없어보이네요.

wheejuni commented on BE/src/main/resources/application-dev.properties:
```diff
@@ -0,0 +1,22 @@
+# ec2 mySQL 연결
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+spring.datasource.url=jdbc:mysql://localhost:3306/airbnb02
+spring.datasource.username=airbnb02
+spring.datasource.password=airbnb02
```
다음 스코프에선 DB 접속정보 암호화까지 도전해보시는 건 어떨까요?

sigridjineth replied:
Jasypt였나요? 그 라이브러리를 Dion이 쓰는 거 봤는데요. 다음 주에는 도입해보도록 하겠습니다.

wheejuni reviewed (commented):
불과 1주일 작업한 내용이라곤 믿기지 않을 정도로 작업량이 많네요.
코딩이 아무리 재밌어도 식사 거르지 마시고요, 즐겁게 꾸준히 하세요. 👍 💯 
ERD에 대한 리뷰는 별도 댓글로 드릴게요. 

아울러 DDD에 대한 부분은 다음 리뷰 요청때 코멘트 드릴 수 있을 것 같아요. 아직 거기까지 작업이 미치지 못하신 것 같아서...
질문사항은 대댓글로 남겨주세요. 감사합니다.

wheejuni commented:
ERD도 살펴봤습니다. 좋네요. 흠잡을 데가 없어요.
관계형DB 특성 잘 살리셔서 설계하신 것 같아요. 이대로 잘 구현되면 좋겠네요. 
뭔가 신랄한 리뷰를 기대하셨을 것 같아 몇번 다시 봤습니다만 제 눈엔 딱히 결점이 안 보여요. 
음.... 뭐라 마무리하지 

화이팅!

sigridjineth commented:
@wheejuni 저는 Room의 세부 정보를 별도 테이블로 빼는 것이 낫지 않겠냐 생각했는데요. 일단 Henry와의 조율 과정에서 다 합쳐두긴 했습니다. 이번에 DDD Quickly인가 책을 보고 있는데 Longitude, Latitude는 하나의 Location이라는 테이블로 묶고 1:1로 연결하더라구요. 뭔가 Room에 정보가 다 으악하고 있는 게 아니라 테이블을 객체처럼 분리해야 하지 않을까 생각이 들었는데 어떻게 보시는지요?

그리고 먼저 스키마를 그리고 객체를 디자인하는 것이 옳은 접근인지 잘 모르겠습니다. DDD에서는 스키마를 먼저 그리는 것이 좋지는 않을 것 같은데, 브라이언의 의견은 어떠신가요?

wheejuni commented:
@jypthemiracle 음 좋은 접근이네요. 그러나 우리는 지금 JPA를 사용하고 있기 때문에, JPA가 제공해주는 객체-테이블 매핑 애노테이션을 충분히 활용하는 것으로 일단 접근해보면 좋을 것 같아요. 

Room의 세부 정보와 관련해서 제가 학습 및 적용 권해드리는 애노테이션은 `@Embeddable` 이에요. 물리적으로는 Room 테이블에 모든 칼럼이 수록되게 되겠지만, 코드 단에서 별도의 객체를 형성해 필드로 관리할 수 있게 됩니다. 그것 자체로도 충분히 객체지향적인 설계를 하면서 테이블 분리는 하지 않아도 됩니다. 

스키마를 그리고 객체를 디자인하는 것 좋은 접근입니다. 다만 "DBMS의 언어" 가 아닌 "Java의 언어" , "OOP의 언어" 로 스키마를 설계했으면 좋겠네요. 그리고 ERD가 아닌 UML을 그려볼 것을 권합니다. 
http://www.nextree.co.kr/p6753/

사실 이 부분은 정답은 없는 주제라서 각자의 생각이 다르고 서로의 의견이 다릅니다. 의견 교류는 활발히 하되 방향성은 뚜렷하게 수립해 놓고 프로젝트 진행하면, 팀 프로젝트 진행간에 더 수월하지 않을까 생각되네요.

sigridjineth commented:
@wheejuni ```@Embedded``` 와 ```@Embeddable``` 학습을 좀 더 해봐야 겠네요. 공부할 게 너무 많은 :>
테이블을 많이 나누는, 이걸 정규화라고 하나요? 정규화가 보통 권장되는 지(= 테이블을 많이 나누는 것이 권장되는 지) 궁금합니다. 상황마다 다르겠지만요.

wheejuni commented:
@jypthemiracle ㅋㅋㅋ 이건 정말 상황마다 다른
도메인의 복잡성에 따라 다르고요, 고용량/초저지연이 필요한 서비스인지, 아니면 그런 부분은 좀 희생해도 되는 서비스인지에 따라 다르고 심지어 어떤 RDBMS 클라우드 솔루션을 쓰고 있는지에 따라 다르고요(어떤 서비스는 테이블 수를 제한하기도 해서....) 

정규화가 필요한 상황, 필요하지 않은 상황 다 다르지만 어쨌든 필요할 때 능숙하게 할 순 있어야 하니까요, 정규화하는 학습과 훈련도 충분히 해 보시기를 권합니다. 

수학적 지식이 약간 필요한 책이긴 하지만 [관계형 데이터베이스 실전 입문](http://www.kyobobook.co.kr/product/detailViewKor.laf?ejkGb=KOR&mallGb=KOR&barcode=9791158390372&orderClick=LAG&Kc=) 이라는 책을 읽어보시는 것도 좋습니다.

sigridjineth commented:
```
2.2 릴레이션의 연산과 술어논리
___제한(Restrict)
___곱집합(Product)
___결합(Join)
___교집합(Intersect)
___합집합(Union)
___차집합(Difference)
```

오 수학의 정석 1단원이 항상 집합이었길래 집합 공부만큼은 많이 했습니다 (꼭 1단원만 보고;;;;;;)

아무튼 항상 정규화를 하려는 습관이 몸에 베어 있었는데 꼭 그런 것은 아니군요. 아직은 판단할 단계에 있지 않으니 프로젝트 하면서 페어랑 잘 맞춰보는 방향으로 가야겠습니다 ㅋㅋ

wooody92 committed:
Merge pull request #39 from codesquad-member-2020/BE/chore/flyway

[BE] DB 형상관리를 위한 Flyway 플러그인 도입

wooody92 committed:
[#45] refactor : ReviewDto 메서드 naming 변경

- ReviewDto.java : create 메서드 naming 변경햇습니다.

wooody92 committed:
[#45] refactor : RoomResponseDto Builder type으로 변경

- RoomResponseDto.java : create 메서드 builder type으로 변경했습니다.

wooody92 committed:
[#45] chore : flyway 적용

- application-dev.properties : local DB에 flyway 적용했습니다.

wooody92 committed:
[#47] refactor : Embedded Location class 분리

- Location.java : Location 클래스 객체를 분리했습니다.

wooody92 committed:
[#45] chore : @Notnull package 변경

- 의존성 추가 후 @Notnull package 변경했습니다.

wooody92 committed:
[#50] feat : Github OAuth를 위한 기본 클래스 생성

- Github OAuth 요청 양식에 필요한 기본 클래스 생성했습니다.

wooody92 committed:
[#50] feat : Github OAuth 기능 추가

- LoginService.java : OAuth 로그인 시 쿠키로 유저의 아이디를 가져오도록 기능 추가했습니다.

wooody92 committed:
[#52] feat : 숙소 리스트 필터링을 위한 query parameter 작성

- 날짜, 인원, 가격, 위치 정보를 갖는 query parameter 작성했습니다.

wooody92 committed:
[#52] feat : 위치 정보 필터링 기능 추가

- RoomService.java : findByLocation location 정보를 받아 일치하는 정보만 출력합니다.

wooody92 committed:
[#52] fix : schema locale 컬럼명 변경 에러 변경

- locale -> location 롤백

wooody92 committed:
[#52] feat : 전체 숙소 리스트와 필터링 된 숙소 리스트를 가져오는 메서드 분리

- RoomController.java : 숙소 리스트를 가져오는 메서드 분리했습니다.

sigridjineth committed:
[#60] chore: flyway user and password modified, DEBUG level

sigridjineth committed:
[#60] refactor: SQL schema to change with field name of Type

sigridjineth committed:
[#60] refactor: RoomType to change with getting name directly from schema

sigridjineth committed:
[#60] refactor: getType method of RoomType to getName re-naming

sigridjineth committed:
[#52] RoomService and RoomController to filter with input priceMin/priceMax queries

- defaultValue of priceMin is 0

sigridjineth committed:
[#52] Room entity to filter according to input priceMin and priceMax

sigridjineth committed:
chore: temporal commit for working price range search

wooody92 committed:
[#52] feat : checkin, checkout LocalDate 형식으로 받도록 변경

- RoomController.java : @DateTimeFormat 사용하여 LocalDate 형식으로 받도록 변경

wooody92 committed:
[#52] fix : primary key 오류로 schema에 p.k 추가

- schema.sql : Pk 추가했습니다.

wooody92 committed:
[#52] feat : checkin - checkout 필터링 기능 추가

- Booking.java : isAvailable : 해당 범위 내에 예약 가능한지 판별하는 메서드 추가했습니다.
- Room.java : 해당 숙소가 갖고 있는 예약된 날짜를 전부 조회하는 메서드 추가했습니다.

wooody92 committed:
[#52] refactor : Location 필터링 기능 객체지향에 맞도록 리팩토링

- Room.java : isValidLocation : location Null 체크 추가
- Locale.java : isEqualsLocation : 입력받은 위치가 존재할 때만 true 반환

wooody92 committed:
Merge pull request #63 from codesquad-member-2020/BE/feat/henry

[BE] 날짜 필터 기능 추가 Henry 작업

wooody92 committed:
Merge pull request #67 from codesquad-member-2020/BE/feat/#52

[BE] 쿼리 파라미터 사용하여 필터링 기능 추가
